SirknightMike/WOT_API
# feat(match): adding match participants and competition tables also co…

### PR Description:
Adding the match participants and competition tables entity and tables in. This PR also contains the new design pattern of service layered. 

### QA Steps:
1. We should make sure that the migrations can run and the table competition, participants, match and roles are added.
2. Confirm that competitions, matches and participants can be added.
3. Create user types in the system. 
4. Permissions should be created.


### Notes: 

None
